### PR TITLE
tests: mem protection tests updated tags

### PR DESCRIPTION
**Pull request by maksimmasalski**
Update current tests Doxygen tags to make them more informative for tests:
-test_mslab()
-test_create_alt_thread()
-test_sys_heap_mem_pool_assign()

For QA is very important to understand goal of each test. I think, adding informative Doxygen tags to the test will help developers to understand Zephyr testing system better.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

**Comment by maksimmasalski** [comment]
@nashif @andyross @andrewboie There is no serious code logic changes, please review changes in Doxygen comments.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016 Intel Corporation
+ * Copyright (c) 2020 Intel Corporation
```
Comment (dcpleung) [inline tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c:2]: Don't replace the year in copyright, but add to it. For example, `Copyright (c) 2016, 2020 Intel Corporation`.

**Review by dcpleung — CHANGES_REQUESTED** [review]

**Review by andyross — COMMENTED** [review]
Might consider calling out "Doxygen links" or something in the title.  "test tags" usually refers to the YAML syntax in test cases used by filtering rules.

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c (hunk 1 above)
Comment (maksimmasalski) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c:2]: Understood, I didn't know that.

**Comment by maksimmasalski** [comment]
> Might consider calling out "Doxygen links" or something in the title. "test tags" usually refers to the YAML syntax in test cases used by filtering rules.

Thanks, I will keep in mind.

**Review by dcpleung — DISMISSED** [review]
The other copyright headers also need to be added, and not replaced.

**Review by nashif — APPROVED** [review]

**Pull request merged**